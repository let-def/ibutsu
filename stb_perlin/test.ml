[@@@ocaml.warning "-6"]
open Stb_perlin

let pf f =
  print_float f;
  print_newline ()

let () = (
  print_endline "Stb_perlin.noise3";
  pf (noise3 0.0 2.0 4.0 1 1 1);
  pf (noise3 0.1 2.0 4.0 1 2 1);
  pf (noise3 0.1 2.1 4.0 1 2 4);
  pf (noise3 0.1 2.1 4.1 4 2 1);
)

let () = (
  print_endline "Stb_perlin.ridge_noise3";
  pf (ridge_noise3 0.0 2.0 4.0 1.0 1.0 0.0 1);
  pf (ridge_noise3 0.1 2.0 4.0 1.0 1.0 0.0 1);
  pf (ridge_noise3 0.1 2.1 4.0 1.0 1.0 0.0 1);
  pf (ridge_noise3 0.1 2.1 4.1 1.0 1.0 0.0 1);
)

let () = (
  print_endline "Stb_perlin.fbm_noise3";
  pf (fbm_noise3 0.0 2.0 4.0 1.0 1.0 1);
  pf (fbm_noise3 0.1 2.0 4.0 1.0 1.0 1);
  pf (fbm_noise3 0.1 2.1 4.0 1.0 1.0 1);
  pf (fbm_noise3 0.1 2.1 4.1 1.0 1.0 1);
)

let () = (
  print_endline "Stb_perlin.turbulence_noise3";
  pf (turbulence_noise3 0.0 2.0 4.0 1.0 1.0 1);
  pf (turbulence_noise3 0.1 2.0 4.0 1.0 1.0 1);
  pf (turbulence_noise3 0.1 2.1 4.0 1.0 1.0 1);
  pf (turbulence_noise3 0.1 2.1 4.1 1.0 1.0 1)
)
