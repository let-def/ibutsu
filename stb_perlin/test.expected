Stb_perlin.noise3
0.
0.
-0.106847889721
-0.189873740077
Stb_perlin.ridge_noise3
0.
0.00570824788883
0.0222782101482
0.0217209588736
Stb_perlin.fbm_noise3
0.
0.10684800148
0.21108391881
0.208427250385
Stb_perlin.turbulence_noise3
0.
0.10684800148
0.21108391881
0.208427250385
